contracts/RelayHub.sol

```diff
@@ -288,42 +334,36 @@ contract RelayHub is IRelayHub {
         // A relay may use a higher gas price than the one requested by the signer (to e.g. get the transaction in a
         // block faster), but it must not be lower. The recipient will be charged for the requested gas price, not the
         // one used in the transaction.
-        require(relayRequest.callData.gasPrice <= tx.gasprice, "Invalid gas price");
-
-        // This transaction must have enough gas to forward the call to the recipient with the requested amount, and not
-        // run out of gas later in this function.
-        require(
-            initialGas >= SafeMath.sub(requiredGas(relayRequest.callData.gasLimit), GAS_OVERHEAD),
-            "Not enough gasleft()");
-
-        address sponsor = relayRequest.relayData.gasSponsor;
-        // We don't yet know how much gas will be used by the recipient, so we make sure there are enough funds to pay
-        // for the maximum possible charge.
-        require(
-            maxPossibleCharge(
-                relayRequest.callData.gasLimit,
-                relayRequest.callData.gasPrice,
-                relayRequest.relayData.pctRelayFee
-            ) <= balances[sponsor],
-            "Sponsor balance too low");
-
-        bytes4 functionSelector = LibBytes.readBytes4(relayRequest.callData.encodedFunction, 0);
-
+        require(relayRequest.gasData.gasPrice <= tx.gasprice, "Invalid gas price");
         bytes memory recipientContext;
-        // We now verify the legitimacy of the transaction (it must be signed by the sender, and not be replayed),
-        // and that the recpient will accept to be charged by it.
-        uint256 preconditionCheck;
-        (preconditionCheck, recipientContext) = canRelay(relayRequest, signature, approvalData);
-
-        if (preconditionCheck != uint256(PreconditionCheck.OK)) {
-            emit CanRelayFailed(
-                msg.sender,
-                relayRequest.relayData.senderAccount,
-                relayRequest.callData.target,
-                sponsor,
-                functionSelector,
-                preconditionCheck);
-            return;
+        GSNTypes.SponsorLimits memory sponsorGasLimits;
+        {
+            uint256 maxPossibleGas;
+            (maxPossibleGas, sponsorGasLimits) = getValidGasLimits(relayRequest.gasData, relayRequest.relayData.gasSponsor);
```

Comment:
The name `getValidGasLimits` is confusing, since it's not only a getter, but also a validator, which its name doesn't necessarily imply ( I understand that it contains "valid" but it still looks like just a getter).

Reply:
Can you come up with a good name for it?